sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,221 @@
+"""Stacking API"""
+
+# Author: Caio Oliveira <caioaao@gmail.com>
+# License: BSD 3 clause
+
+from ..base import (BaseEstimator, TransformerMixin, MetaEstimatorMixin)
+from ..model_selection import cross_val_predict
+from ..pipeline import FeatureUnion
+from ..preprocessing import FunctionTransformer
+
+
+class StackingTransformer(BaseEstimator, MetaEstimatorMixin, TransformerMixin):
+    """Transformer to turn estimators into meta-estimators for model stacking
+
+    In stacked generalization, meta estimators are combined in layers to
+    improve the final result. To prevent data leaks between layers, a procedure
+    similar to cross validation is adopted, where the model is trained in one
+    part of the set and predicts the other part. In ``StackingTransformer``, it
+    happens during ``fit_transform``, as the result of this procedure is what
+    should be used by the next layers. Note that this behavior is different
+    from ``fit().transform()``. See the :ref:`User Guide <ensemble>` for more
+    info.
+
+    Parameters
+    ----------
+    base_estimator : the estimator to be blended.
+
+    cv : cv to be used, optional (default=3)
+        Will be passed to ``cross_val_predict`` during ``fit_transform``.
+
+    method : string, optional (default='auto')
+        Invokes the passed method name of the passed estimator. If the method
+        is ``auto``, will try to invoke, for each estimator, ``predict_proba``,
+        ``decision_function`` or ``predict`` in that order.
+
+    n_jobs : int, optional (default=1)
+        Number of jobs to be passed to ``cross_val_predict`` during
+        ``fit_transform``.
+
+    """
+    def __init__(self, base_estimator, cv=3, method='auto', n_jobs=1):
+        self.base_estimator = base_estimator
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y, **fit_params):
+        """Fit the estimator.
+
+        This should only be used in special situations. See :ref:`user guide`
+        for more info.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        **fit_params : parameters to be passed to the base estimator.
+
+        Returns
+        -------
+        self : object
+
+        """
+        self.base_estimator.fit(X, y, **fit_params)
```

Comment:
we need to clone this and fit the cloned instance, e.g. `self.base_estimator_`. Perhaps drop the "base_" too.